rfc/ipni-w3c.md

```diff
@@ -0,0 +1,346 @@
+
+After storing the CAR and CARx-index, w3s creates a [location claim](https://hackmd.io/@gozala/content-claims?utm_source=preview-mode&utm_medium=rec#Location-Claims) for the CAR file and the CARx-index. Each location claim maps an item's CID to a URL where the item can be retrieved.
+
+w3s creates a Content Claims Bucket. This is a container, stored on w3s network, that holds all content claims associated with the stored data. The location claims and inclusion claim are written to the claims bucket. The claims bucket is then stored.
+
+[IPNI](https://github.com/ipni/specs/blob/main/IPNI.md) will allow a client to lookup any of the multihashes, that are in the CARx-index, to get information sufficient to recover the claims bucket. The location claims from the claims bucket can then be used to get the CARx-index and CAR data.
+
+Information is provided to IPNI in the form of [IPNI Advertisements](https://github.com/ipni/specs/blob/main/IPNI.md#advertisements)
+that are signed by w3s and published to IPNI by w3s. Each  Advertisement 
+
+When a new advertisement is published, an [IPNI Announcement](https://github.com/ipni/specs/blob/main/IPNI.md#announcements) is sent to IPNI to trigger IPNI to fetch the new advertisement.
+
+## Introduction
+
+We publish ad-hoc batches of multihashes to IPNI. This proposal aims to align our usage of IPNI with [content-claims], by publishing an advert per [inclusion claim], and include the source claim in the IPNI Advertisement
+
+### Motivation
+
+- Align IPNI advertisement entries with CAR block sets, so that these CIDs can be queried to get the location of the CARx-index and CAR data.
+  - This exposes our block-to-car indexes. Anyone could use IPNI to find which CAR a block is in.
+  -   The context id bytes provide the CAR CID for any block look up. The CAR CID can then be used to find the CAR index via our content-claims API.
+  - We could delete the IPNI records by CAR CID if the CAR is deleted.
+- Make IPNI advertising an explicit UCAN capability that clients can invoke rather than a side-effect of bucket events
+  - With this we are free to write CARs anywhere. The user's agent invokes a `ipni/offer` capability to ask us to publish an IPNI advertisement for the blocks in their CAR.
+  - This empowers the user to opt-in or out as they need, and allows us to bill for the (small) cost of running that service.
+- Provide a way to retrieve a CAR index, that has (CID, offset, length) tuples, without retrieving the CAR file.
+    - Eliminates the need to rely on the CARPark database. Block location information can now be stored on the w3s network.
+    - Clients can request specific data item and only receive data for that item.
+- Put the source inclusion claim in the IPNI advert metadata.
+  - We have to sign IPNI Adverts as the provider. Providing a signed source claim allows more nuanced reputation decisions.
```

Comment:
Could the source inclusion claim be signed by the client, if IPNI supported UCAN delegations?

Reply:
Precisely this ! If we delegate user claim signed by us, it’s up to them to decide who they share it with. They can share with everyone or just share with selected group. They can also leverage UCAN expiry and notBefore fields to reduce level of access shared.

Reply:
That was what I was saying, I think I left out a word. it should read:
"Providing a client-signed source claim allows more nuanced reputation decisions."

So the client should sign the source claim.  Also, the client can be the signer of the IPNI advertisement, signing as the provider (using the key that matches the ProviderID).